### PR TITLE
Fix object mock failure for resource methods

### PR DESCRIPTION
**Pull request by Dilhasha**
- Consider resource functions when validating member functions
- Adds check against resource functions during the method signature validation

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/39644

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/testerina-integration-test/src/test/resources/project-based-tests/object-mocking-tests2/Dependencies.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# AUTO-GENERATED FILE. DO NOT MODIFY.
```
Comment (gayaldassanayake) [inline tests/testerina-integration-test/src/test/resources/project-based-tests/object-mocking-tests2/Dependencies.toml:1]: Do we need `.gitignore`, `Dependencies.toml` files to be comited here?

Comment (Dilhasha) [reply tests/testerina-integration-test/src/test/resources/project-based-tests/object-mocking-tests2/Dependencies.toml]: I'll remove those files.

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java [hunk 2]
```diff
@@ -95,6 +95,23 @@ public void testObjectMocking() throws BallerinaTestException {
...
+     * @throws BallerinaTestException
+     */
+    @Test()
+    public void testObjectMockDouble() throws BallerinaTestException {
+        String msg1 = "1 passing";
+        String[] args = mergeCoverageArgs(new String[]{"object-mocking-tests2"});
+        String output = balClient.runMainAndReadStdOut("test", args,
+                new HashMap<>(), projectPath, false);
+        if (!output.contains(msg1)) {
+            Assert.fail("Test failed due to object mocking failure with resource functions in test framework." +
```
Comment (Thevakumar-Luheerathan) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java:110]: Can we store the complete output in a file and carry out the test against it??

**Review by Thevakumar-Luheerathan — APPROVED** [review]

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java (hunk 2 above)
Comment (Dilhasha) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java:110]: Decided to address this with https://github.com/ballerina-platform/ballerina-lang/pull/39562

Comment (Thevakumar-Luheerathan) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java:110]: We will address this comment in https://github.com/ballerina-platform/ballerina-lang/issues/39561.

**Pull request merged**